lib/settings.js

```diff
@@ -424,12 +424,51 @@ ${this.results.reduce((x, y) => {
   async getRepoConfigMap () {
     try {
       this.log.debug(` In getRepoConfigMap ${JSON.stringify(this.repo)}`)
+      // GitHub getContent api has a hard limit of returning 1000 entries without
+      // any pagination. They suggest to use Tree api.
+      // https://docs.github.com/en/rest/repos/contents?apiVersion=2022-11-28#get-repository-content
+
+      // get .github/repos directory sha to use in the getTree api
       const repo = { owner: this.repo.owner, repo: 'admin' }
-      const CONFIG_PATH = '.github'
-      const params = Object.assign(repo, { path: path.posix.join(CONFIG_PATH, 'repos'), ref: this.ref })
+      const params = Object.assign(repo, { path: path.posix.join(CONFIG_PATH), ref: this.ref })
+      const githubDirectoryContentResponse = await this.github.repos.getContent(params).catch(e => {
+        this.log.debug(`Error getting settings ${JSON.stringify(params)} ${e}`)
+      })
 
-      const response = await this.loadConfigMap(params)
-      return response
+      if (!githubDirectoryContentResponse) {
+        throw new Error('Error reading .github directory')
+      }
+
+      const githubDirContent = githubDirectoryContentResponse.data
+      const repoDirInfo = githubDirContent.filter(dir => dir.name === 'repos')[0]
+      if (!repoDirInfo) {
+        this.log.debug('No repos directory in the admin/.github')
+        return []
+      }
+
+      // read the repo contents using tree
+      this.log.debug(`repos directory info ${JSON.stringify(repoDirInfo)}`)
+      const endpoint = `/repos/${this.repo.owner}/${repo.repo}/git/trees/${repoDirInfo.sha}`
```

Comment:
One option would be to replace `github.request` with some syntactic sugar `github.git.getTree({owner, repo, tree_sha}) ` https://octokit.github.io/rest.js/v18#git-get-tree